Prepare kubeflow manifest for release

lucferbux opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- [UI] Include any screenshots of changed UI; Include any gifs if it was a flow / UX change -->
Prepare kubeflow manifests for release adding a kubeflow manifest and allowing tagging in images

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
**Note: I still need to test the tag in my local environment**
Test deployment just following the instructions

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [X] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work.
- [X] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

<!--- You can ignore these if you are doing Go Model Registry REST server, MR Python client, manifest, controller, internal logic, etc changes; aka non-UI / visual changes -->
- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

ederign commented:
Great work @lucferbux ! I've tested this locally and got it working after apply https://github.com/kubeflow/manifests/pull/2989 !

![Screenshot 2025-02-12 at 10 44 33 AM](https://github.com/user-attachments/assets/43f20c2f-eee5-430f-862f-3c6784e080d9)

ederign commented:
Also, from /ui make kubeflow-deployment worked for me! @lucferbux as a FUP PR, we should add this option to UI readme.

ederign commented:
/approve

ederign commented:
/lgtm

ederign commented:
@tarilabs @rareddy, As this touches manifests and actions, I don't have the right to merge it.

btw, the only part that I didn't test was the github action!

tarilabs reviewed (commented):
@ederign @lucferbux overall LGTM but I believe the instruction "stable" should PATCH the existing KF Dashboard ConfigMap; am I right? Otherwise, newer modification to the CM by the platform would be elided by this version in its entirety.

Please log this as a TODO task.

Then, I'm happy to merge this PR (and the TODO task github issue could be dealt with separately)

wdyt?

lucferbux commented:
> @ederign @lucferbux overall LGTM but I believe the instruction "stable" should PATCH the existing KF Dashboard ConfigMap; am I right? Otherwise, newer modification to the CM by the platform would be elided by this version in its entirety.
> 
> Please log this as a TODO task.
> 
> Then, I'm happy to merge this PR (and the TODO task github issue could be dealt with separately)
> 
> wdyt?

That's a great suggestion, and yeah, I just copied the overlay of kserve but then I realize they have in in centraldashboard which is weird, https://github.com/kubeflow/manifests/blob/master/apps/centraldashboard/upstream/overlays/kserve/patches/configmap.yaml

Let me rework that and test the tag release and it would be ready to go, thanks a log @tarilabs

tarilabs commented:
> Let me rework that and test the tag release and it would be ready to go, thanks a log @tarilabs

it's good with me if you want to work on this now, or later. Up to you when is most suitable. Just lmk again when time to merge this PR

lucferbux commented:
Ok a couple of things:

1. I've removed the `kubeflow` overlay, got `istio` as the main one added the kubeflow namespace as an opinionated install and changed all the docs to point at that
2. I've removed the configmap with that, I've added instructions on how to add that in a one line command (i think it's the best and we can add that into the install docs too)
3. I've tested the tag release in my fork and it just works great (pointing at my registry)
<img width="1961" alt="Screenshot 2025-02-12 at 18 03 27" src="https://github.com/user-attachments/assets/710259de-bc12-447c-8100-8c1ed3062b6b" />
<img width="1096" alt="Screenshot 2025-02-12 at 18 03 19" src="https://github.com/user-attachments/assets/804f227d-e0e9-4991-a584-a642aec80dad" />

tarilabs reviewed (commented):
thanks @lucferbux 

/lgtm

@ederign ?

ederign commented:
@tarilabs @lucferbux I did the installation again and can confirm that it works. (I didn't tested the gh action)

ederign commented:
/approve

tarilabs reviewed (approved):
/approve
